sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Customizations/ContainerRegistryCredentials.cs

```diff
@@ -280,29 +281,31 @@ private string GetScopeFromHeaders(HttpHeaders headers)
                 if (headerKVP.Key.ToLower() == challengeHeader)
                 {
                     headerValue = string.Join(",", headerKVP.Value);
-                    
+
                     break;
                 }
             }
 
-            string scope = "";
             int position = headerValue.IndexOf("scope=");
-            if (position > 0)
+            string scope = headerValue.Substring(position);
+            string[] keyValues = scope.Split('=');
+            int length = keyValues.Length;
+            var scopeContainedIn = keyValues[1];
+
+            if (length < 2)
             {
-                scope = headerValue.Substring(position);
+               throw new Exception($"Could not find a scope in the {headerValue}");
             }
-            else
+            else if(length > 2)
             {
-                throw new Exception($"Could not find a scope in the {headerValue}");
+                var pattern = "\".+:.+:.+\"";
+                var regex = new Regex(pattern);
```

Comment:
It would be better to make `pattern` a `const` and `regex` and a `private static readonly` member.